src/test/java/org/opentripplanner/middleware/triptracker/TripSurveyUploadJobTest.java

```diff
@@ -0,0 +1,178 @@
+
+        surveyUploads = Lists.newArrayList(surveyUploadTwoDaysAgo, surveyUploadThreeDaysAgo);
+
+        Persistence.tripSurveyUploads.create(surveyUploadTwoDaysAgo);
+        Persistence.tripSurveyUploads.create(surveyUploadThreeDaysAgo);
+    }
+
+    @AfterAll
+    public static void tearDown() {
+        assumeTrue(IS_END_TO_END);
+        
+        // Delete trip survey upload entries
+        surveyUploads.forEach(upload -> {
+            TripSurveyUpload storedUpload = Persistence.tripSurveyUploads.getById(upload.id);
+            if (storedUpload != null) {
+                Persistence.tripSurveyUploads.removeById(storedUpload.id);
+            }
+        });
+    }
+    
+    @AfterEach
+    void afterEach() throws Exception {
+        assumeTrue(IS_END_TO_END);
+
+        // Delete trip survey upload entries that were added.
+        for (TripSurveyUpload upload : Persistence.tripSurveyUploads.getAll()) {
+            if (!(upload.id.equals(surveyUploadTwoDaysAgo.id) && !upload.id.equals(surveyUploadThreeDaysAgo.id))) {
+                Persistence.tripSurveyUploads.removeById(upload.id);
+            }
+        }
+
+        // Delete any files we created
+        if (tempFile != null) {
+            FileUtils.deleteFile(tempFile);
+            tempFile = null;
+        }
+        if (zipFileName != null) {
+            S3Utils.deleteObject(
+                ConnectedDataManager.CONNECTED_DATA_PLATFORM_S3_BUCKET_NAME,
+                String.format("%s/%s", ConnectedDataManager.CONNECTED_DATA_PLATFORM_S3_FOLDER_NAME, zipFileName)
+            );
+            zipFileName = null;
+        }
+    }
+
+    /**
+     * Make sure that the first upload is created and contains the correct upload date.
+     */
+    @Test
+    void canStageFirstUpload() {
+        // Just for this test, delete the previous days.
+        Persistence.tripSurveyUploads.removeById(surveyUploadTwoDaysAgo.id);
+        Persistence.tripSurveyUploads.removeById(surveyUploadThreeDaysAgo.id);
+
+        TripSurveyUploadJob job = new TripSurveyUploadJob();
+        job.stageUploadDays();
+        TripSurveyUpload upload = IntervalUpload.getFirst(Persistence.tripSurveyUploads);
+        assertNotNull(upload);
+        assertTrue(PREVIOUS_WHOLE_DAY_FROM_NOW.isEqual(upload.uploadHour));
+    }
+
+    @Test
+    void canRunJob() {
+        assumeTrue(IS_END_TO_END);
+
+        // After running the job, assuming that API calls were successful,
+        // the upload records should be updated.
+
+        TripSurveyUploadJob job = new TripSurveyUploadJob(localDateTime -> createSurveyApiResponse(), () -> EXPECTED_HEADER);
+        job.run();
+
+        TripSurveyUpload upload = job.getLastUploadCreated();
+        assertNotNull(upload);
+        assertTrue(PREVIOUS_WHOLE_DAY_FROM_NOW.isEqual(upload.uploadHour));
+        assertEquals(IntervalUploadStatus.COMPLETED, upload.status);
+    }
+
+    @Test
+    void canMakeResponsesParams() {
```

Comment:
This fails for me and is no doubt time zone related:

`expected: <?page_size=1000&since=1729839600&until=1729925999> but was: <?page_size=1000&since=1729810800&until=1729897199>`

Reply:
On CI, the timezone is America/Los_Angeles (so set your env.yml for E2E tests to that time zone). I added clarifying comments in https://github.com/ibi-group/otp-middleware/pull/267/commits/79b4c7aa5d81c03eb8447d53e89ab8a060e52f83.